app/charts/map/chart-map-prototype.tsx

```diff
@@ -65,74 +55,78 @@ export const ChartMapVisualization = ({
-        rawShapes.objects.shapes
-      ) as GeoJSON.FeatureCollection<
-        GeoJSON.GeometryObject,
-        ShapeFeature["properties"]
-      >;
-
-      topojsonShapes.features.forEach((d) => {
-        // Should we match by labels?
-        d.properties.observation = observations?.find(
-          (o) => o[areaLayerComponentIri] === d.properties.label
-        );
-      });
-
-      return {
-        shapes: topojsonShapes,
-        mesh: topojsonMesh(rawShapes, rawShapes.objects.shapes as any),
-      };
+    if (dimensions) {
+      const geoShapesDimensions = getGeoShapesDimensions(dimensions);
+
+      if (geoShapesDimensions.length) {
+        const areaGeoDimension = dimensions?.find(
+          (d) => d.iri === areaGeoDimensionIri
+        ) as GeoShapesDimension | undefined;
+
+        if (areaGeoDimension) {
+          const rawShapes = areaGeoDimension.geoShapes as GeoShapes;
+          const topojsonShapes = topojsonFeature(
+            rawShapes,
+            rawShapes.objects.shapes
+          ) as GeoJSON.FeatureCollection<
+            GeoJSON.GeometryObject,
+            ShapeFeature["properties"]
+          >;
+
+          topojsonShapes.features.forEach((d) => {
+            // Should we match by labels?
+            d.properties.observation = observations?.find(
+              (o) => o[areaGeoDimensionIri] === d.properties.label
+            );
+          });
+
+          return {
+            shapes: topojsonShapes,
+            mesh: topojsonMesh(rawShapes, rawShapes.objects.shapes as any),
+          };
+        }
+      }
     }
-  }, [areaLayerComponentIri, areaLayerDimension, observations]);
+  }, [areaGeoDimensionIri, dimensions, observations]);
 
   const symbolLayer = useMemo(() => {
     if (dimensions) {
       const geoCoordinatesDimensions = getGeoCoordinatesDimensions(dimensions);
 
       if (geoCoordinatesDimensions.length) {
         const geoCoordinatesDimension = dimensions.find(
-          (d) => d.iri === symbolLayerComponentIri
+          (d) => d.iri === symbolGeoDimensionIri
         ) as GeoCoordinatesDimension | undefined;
 
         if (geoCoordinatesDimension) {
           return (
             geoCoordinatesDimension.geoCoordinates as RawGeoCoordinates[]
-          ).map(
-            (d) =>
-              ({
-                coordinates: [+d.longitude!, +d.latitude!],
-                properties: {
-                  iri: d.iri,
-                  label: d.label,
-                  observation: observations?.find(
-                    (o) => o[symbolLayerComponentIri] === d.label
-                  ),
-                },
-              } as GeoPoint)
-          );
+          ).map((d) => ({
+            coordinates: [d.longitude, d.latitude] as [number, number],
+            properties: {
+              iri: d.iri,
+              label: d.label,
+              observation: observations?.find(
+                (o) => o[symbolGeoDimensionIri] === d.label
+              ),
```

Comment:
A observations by label could be useful. Since it is needed in a lot of places, I wonder if we should have a DataProvider somewhere in the tree that does the indexing so that children do not have to do this repeatedly, what do you think ? (this is not the scope of the PR).